### PR TITLE
roachtest: Index overload automation of TPC-E

### PR DESCRIPTION
**Pull request by andrewbaptist**
This patch adds the ability to run tests against n2 standard from within roachprod / roachtest.

Release note: None
Epic: None

### CODE REVIEW

**Review by irfansharif — COMMENTED** [review]
I know that @dt's working on adding roachtest libraries to snapshot GCE images and use them for quicker turn around times. He's also actually doing it for TPC-E restores I think. This PR should wait until that gets done -- this is far too long running and $$$ to be useful otherwise.

**Comment by andrewbaptist** [comment]
What if we merged it with the skip test flag - since the ability to run skipped tests is about to merge

File: pkg/cmd/roachtest/test_registry.go [hunk 1]
```diff
@@ -137,7 +139,7 @@ func (r *testRegistryImpl) prepareSpec(spec *registry.TestSpec) error {
 	// may not be scheduled until a few hours in due to the CPU quota, individual
 	// tests should expect to take "less time". Longer-running tests require the
 	// weekly tag.
-	const maxTimeout = 18 * time.Hour
+	const maxTimeout = 7 * 24 * time.Hour
```
Comment (irfansharif) [inline pkg/cmd/roachtest/test_registry.go:142]: Why this change?

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/prometheus"
+	"github.com/cockroachdb/errors"
+)
+
+// TODO(baptist): This shares a lot of code with tpce.go. Consider ways to refactor.
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:29]: Let's do this now.

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+		startOpts := option.DefaultStartOpts()
+		settings := install.MakeClusterSettings(install.NumRacksOption(racks))
+		c.Start(ctx, t.L(), startOpts, settings, roachNodes)
+
+		{
+			promCfg := &prometheus.Config{}
+			promCfg.WithPrometheusNode(workloadNode.InstallNodes()[0])
+			promCfg.WithNodeExporter(c.All().InstallNodes())
+			promCfg.WithCluster(roachNodes.InstallNodes())
+			promCfg.WithGrafanaDashboard("http://go.crdb.dev/p/snapshot-admission-control-grafana")
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:61]: Use https://go.crdb.dev/p/index-admission-control-grafana instead.

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+			case <-importC:
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("waiting before creating the index")
+
+			// Wait for the load to run for a while before starting the index build.
+			select {
+			case <-time.After(opts.delayBeforeIndex):
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:119]: Look at how the TPC-C harness does things. It's very flexible, and better in many regards. There we have that `During` callback where test authors can inject whatever sleep they want during a test run, and this sort of thing would apply. So I would suggest improving the TPC-E test harness first as part of this PR and then adding a good use of it. It'll also encourage more TPC-E roachtests from other teams.

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+			case <-ctx.Done():
+				return ctx.Err()
+			}
+
+			t.Status("creating index (~30min)")
+
+			db := c.Conn(ctx, t.L(), 1)
+			_, err := db.ExecContext(ctx, "CREATE INDEX ON tpce.cash_transaction (ct_dts)")
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:127]: The same suggestion as the other PR -- use an explicit, unique name, suffixed perhaps by the timestamp this was run at so that repeated runs are made easy.

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+				activeRatio:      0.2,
+				threads:          400,
+				delayBeforeIndex: 1 * time.Hour,
+			}
+			runTPCE(ctx, t, c, opts)
+		},
+	})
+
+	// This test takes about 1 hours to run.
+	r.Add(registry.TestSpec{
```
Comment (irfansharif) [inline pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:158]: Meh, let's just have the one test above and make sure it runs <1h. And also, within reason, support either --local and/or --skip-init if you use an already set up cluster to keep re-running things without the bulk import. This TPC-E harness also needs to integrate with prometheus/grafana (I see you've done so manually above), let's do that in this PR -- steal as many ideas from the TPC-C harness when doing so.

**Review by irfansharif — COMMENTED** [review]
I'm ok merging this test as skipped until the GCE snapshotting work but we ought to improve the TPC-E harness first here instead of duplicating a bunch of code (even if this was the easiest thing to do at the start). Left some comments below but the TL;DR is lets make this as good as the TPC-C harness; I think it's doable since it's not a ton of code for either today. Feel free to do this work in whatever PRs/commits make life easiest for you.

File: pkg/cmd/roachtest/test_registry.go (hunk 1 above)
Comment (andrewbaptist) [reply pkg/cmd/roachtest/test_registry.go:142]: Oops - removed this change. thanks for catching this.

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go (hunk 3 above)
Comment (andrewbaptist) [reply pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:61]: Done

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go (hunk 5 above)
Comment (andrewbaptist) [reply pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:127]: I realized when iterating with that one that not specifying a name gives the flexibility to re-run the identical create index (the system does this automatically) - so its not necessary to pick an explicit name unless you want to drop it in the end. Since I wasn't dropping I didn't pick a name. Let me know if you think I should change this.

File: pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go (hunk 6 above)
Comment (andrewbaptist) [reply pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go:158]: I'm not sure I understand this request. I don't think it is possible to create a test that runs <1hr, uses default settings and sees this overload issue (since it needs 1000 L0 files as part of index creation). I wasn't sure how best to approach this, so definitely open to ideas on how to do this (combination of skipped tests, short test that doesn't show the issue or other).

Comment (andrewbaptist) [reply pkg/cmd/roachtest/tests/admission_control_index_overload_tpce.go]: Removed the small test below.

**Comment by irfansharif** [comment]
This is subsumed by https://github.com/cockroachdb/cockroach/pull/103816.
